engine/src/main/java/org/terasology/rendering/primitives/BlockMeshGeneratorSingleShape.java

```diff
@@ -89,35 +89,39 @@ public void generateChunkMesh(ChunkView view, ChunkMesh chunkMesh, int x, int y,
             renderType = ChunkMesh.RenderType.BILLBOARD;
         }
 
-        if (blockAppearance.getPart(BlockPart.CENTER) != null) {
+        BlockMeshPart centerPart = blockAppearance.getPart(BlockPart.CENTER);
+        if (centerPart != null) {
             Vector4f colorOffset = selfBlock.calcColorOffsetFor(BlockPart.CENTER, selfBiome);
-            blockAppearance.getPart(BlockPart.CENTER).appendTo(chunkMesh, x, y, z, colorOffset, renderType, vertexFlag);
+            centerPart.appendTo(chunkMesh, x, y, z, colorOffset, renderType, vertexFlag);
         }
 
         boolean[] drawDir = new boolean[6];
 
         for (Side side : Side.values()) {
-            drawDir[side.ordinal()] = blockAppearance.getPart(BlockPart.fromSide(side)) != null && isSideVisibleForBlockTypes(adjacentBlocks.get(side), selfBlock, side);
+            int sideOrd = side.ordinal();
+            drawDir[sideOrd] = blockAppearance.getPart(BlockPart.fromSide(side)) != null &&
+                    isSideVisibleForBlockTypes(adjacent[sideOrd], selfBlock, side);
         }
 
         // If the selfBlock is lowered, some more faces may have to be drawn
         if (selfBlock.isLiquid()) {
-            Block bottomBlock = adjacentBlocks.get(Side.BOTTOM);
+            Block bottomBlock = adjacent[1 /* BOTTOM */];
```

Comment:
Any way to get a better idea of how much it matters? Or at least use a local constant instead of a magic number + include a good comment explaining the details?